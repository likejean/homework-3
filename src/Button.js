import React from 'react';

const Button = ({ setSubmit, inputs }) => <button data-testid="button" className="w3-btn w3-border w3-hover-blue" onClick={() => setSubmit(inputs)}>SUBMIT</button>
export default Button;