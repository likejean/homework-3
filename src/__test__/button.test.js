import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';


afterEach(cleanup);

it('renders without crushing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
});

it('renders button correctly', () => {
    const {getByTestId} = render(<Button>SUMBIT</Button>);
    expect(getByTestId('button')).toHaveTextContent('SUBMIT');
});


it("matches snapshot-1", () => {
    const tree = renderer.create(<Button>SUBMIT</Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot-2", () => {
    const tree = renderer.create(<Button>SAVE</Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

