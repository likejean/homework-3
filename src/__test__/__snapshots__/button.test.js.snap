// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot-1 1`] = `
<button
  className="w3-btn w3-border w3-hover-blue"
  data-testid="button"
  onClick={[Function]}
>
  SUBMIT
</button>
`;

exports[`matches snapshot-2 1`] = `
<button
  className="w3-btn w3-border w3-hover-blue"
  data-testid="button"
  onClick={[Function]}
>
  SUBMIT
</button>
`;
